metrics/web/test/auth/presentation/state/auth_notifier_test.dart

```diff
@@ -124,43 +228,219 @@ void main() {
+      ".subscribeToAuthenticationUpdates() delegates to receive user profile updates when the user is not null",
+      () {
+        final authNotifier = AuthNotifier(
+          receiveAuthUpdates,
+          signInUseCase,
+          googleSignInUseCase,
+          signOutUseCase,
+          receiveUserProfileUpdates,
+          createUserProfileUseCase,
+          updateUserProfileUseCase,
+        );
+        when(receiveAuthUpdates(any)).thenAnswer((_) => Stream.value(user));
+
+        when(receiveUserProfileUpdates(any)).thenAnswer(
+          (_) => Stream.value(userProfile),
+        );
+
+        final listener = expectAsyncUntil0(() {}, () {
+          if (authNotifier.userProfileModel != null) {
+            verify(receiveUserProfileUpdates(any)).called(1);
+            return true;
+          }
+          return false;
+        });
+
+        authNotifier.addListener(listener);
+
+        authNotifier.subscribeToAuthenticationUpdates();
+      },
+    );
+
+    test(
+      ".subscribeToAuthenticationUpdates() unsubscribes from user updates when a user is null",
+      () {
+        final userController = StreamController<User>();
 
-      when(receiveAuthUpdates()).thenAnswer((_) => Stream.value(user));
+        when(receiveAuthUpdates()).thenAnswer((_) => userController.stream);
 
-      authNotifier.subscribeToAuthenticationUpdates();
+        authNotifier.subscribeToAuthenticationUpdates();
+
+        expect(userController.hasListener, isTrue);
 
-      await authNotifier.signInWithEmailAndPassword(email, password);
+        when(receiveAuthUpdates()).thenAnswer((_) => const Stream.empty());
 
-      expect(authNotifier.isLoggedIn, isTrue);
-    });
+        authNotifier.subscribeToAuthenticationUpdates();
+
+        expect(userController.hasListener, isFalse);
+      },
+    );
+
+    test(
+      ".subscribeToUserProfileUpdates() subscribes to a user profile updates stream",
+      () {
+        final authNotifier = AuthNotifier(
+          receiveAuthUpdates,
+          signInUseCase,
+          googleSignInUseCase,
+          signOutUseCase,
+          receiveUserProfileUpdates,
+          createUserProfileUseCase,
+          updateUserProfileUseCase,
+        );
+        final streamController = StreamController<UserProfile>();
+
+        when(receiveAuthUpdates(any)).thenAnswer(
+          (_) => Stream.value(user),
+        );
+        when(receiveUserProfileUpdates(any)).thenAnswer(
+          (_) => streamController.stream,
+        );
+
+        final listener = expectAsyncUntil0(
+          () {},
+          () => streamController.hasListener,
+        );
+        authNotifier.subscribeToAuthenticationUpdates();
+
+        authNotifier.addListener(listener);
+        streamController.add(userProfile);
+      },
+    );
 
     test(
-      ".isLoggedIn status is true after a user signs in with Google",
+      ".subscribeToUserProfileUpdates() creates a user profile model from the user profile that stream emits",
```

Comment:
```suggestion
      ".subscribeToUserProfileUpdates() creates a user profile model once receiving the user profile",
```
Would it be more clear?